Microsoft.Toolkit.Uwp/Helpers/DispatcherHelper.cs

```diff
@@ -0,0 +1,132 @@
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Core;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+
+namespace Microsoft.Toolkit.Uwp.Helpers
+{
+    /// <summary>
+    /// This class provides static methods helper for executing code in UI thread of the main window.
+    /// </summary>
+    public static class DispatcherHelper
+    {
+        /// <summary>
+        /// Execute the given function asynchronously on UI thread of the main view
+        /// </summary>
+        /// <typeparam name="T">returned data type of the function</typeparam>
+        /// <param name="function">Asynchronous function to be executed asynchronously on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task with type <typeparamref name="T"/></returns>
+        public static async Task<T> ExecuteOnUIThreadAsync<T>(Func<Task<T>> function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            return await ExecuteOnUIThreadAsync<T>(CoreApplication.MainView, function, priority);
+        }
+
+        /// <summary>
+        /// Execute the given function asynchronously on given view's UI thread. Default view is the main view.
+        /// </summary>
+        /// <typeparam name="T">returned data type of the function</typeparam>
+        /// <param name="viewToExecuteOn">View for the <paramref name="function"/>  to be executed on </param>
+        /// <param name="function">Asynchronous function to be executed asynchronously on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task with type <typeparamref name="T"/></returns>
+        public static async Task<T> ExecuteOnUIThreadAsync<T>(CoreApplicationView viewToExecuteOn, Func<Task<T>> function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            return await viewToExecuteOn?.Dispatcher?.AwaitableRunAsync<T>(function, priority);
+        }
+
+        /// <summary>
+        /// Execute the given function asynchronously on given view's UI thread. Default view is the main view.
+        /// </summary>
+        /// <param name="viewToExecuteOn">View for the <paramref name="function"/>  to be executed on </param>
+        /// <param name="function">Asynchronous function to be executed asynchronously on UI thread</param>
+        /// <param name="priority">Dispatcher execution priority, default is normal</param>
+        /// <returns>Awaitable Task</returns>
+        public static async Task ExecuteOnUIThreadAsync(CoreApplicationView viewToExecuteOn, Func<Task> function, CoreDispatcherPriority priority = CoreDispatcherPriority.Normal)
+        {
+            await ExecuteOnUIThreadAsync<bool>(
```

Comment:
Supposedly it is better/cheaper to use `<object>` and return null when you just want to return Task from a Task<T>.
Also no reason to await here. Remove `Async` from message signature and just return the task.
`return ExecuteOnUIThreadAsync<bool>(...);`